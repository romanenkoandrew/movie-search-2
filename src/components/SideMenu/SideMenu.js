import React from 'react';
import PropTypes from 'prop-types';

import { Menu, Layout } from 'antd';
import { CarOutlined, ProfileOutlined } from '@ant-design/icons';
import { Link } from 'react-router-dom';
import styles from './SideMenu.styles';
import { MAIN_ROUTE, WATCHLIST_ROUTE } from 'constants/routing';

const { Sider } = Layout;

const SideMenu = () => {
  const [isMenuOpen, setIsMenuOpen] = React.useState(false);
  const toggleMenu = () => {
    setIsMenuOpen(!isMenuOpen);
  };
  return (
    <Sider
      css={styles.sider}
      theme='light'
      collapsible
      collapsed={isMenuOpen}
      onCollapse={toggleMenu}
    >
      <Menu mode='inline' theme='light' css={styles.menuItem}>
        <Menu.Item key='main_drivers' icon={<CarOutlined />}>
          <Link to={MAIN_ROUTE}>Main</Link>
        </Menu.Item>
        <Menu.Item key='main_orders' icon={<ProfileOutlined />}>
          <Link to={WATCHLIST_ROUTE}>WatchList</Link>
        </Menu.Item>
      </Menu>
    </Sider>
  );
};

SideMenu.propTypes = {};

export default SideMenu;
