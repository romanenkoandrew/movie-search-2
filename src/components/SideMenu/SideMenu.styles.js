import { css } from '@emotion/core';

export default {
  sider: () => css`
    border-right: 1px solid rgba(171, 171, 171, 0.8);
  `,
};
