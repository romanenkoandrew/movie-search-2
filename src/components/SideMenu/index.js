import SideMenu from './SideMenu';
import { connect } from 'react-redux';
import selector from './selector';
import { toggleSideMenu } from 'actions';

const mapDispatchToProps = { toggleSideMenu };

export default connect(selector, mapDispatchToProps)(SideMenu);
