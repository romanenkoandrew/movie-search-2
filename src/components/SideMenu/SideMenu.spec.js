import React from 'react';
import { shallow } from 'enzyme';
import SideMenu from './SideMenu';

describe('SideMenu component', () => {
  it('should render SideMenu component', () => {
    const component = shallow(
      <SideMenu sideMenuIsOpen toggleSideMenu={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it('should render SideMenu component', () => {
    const component = shallow(
      <SideMenu sideMenuIsOpen={false} toggleSideMenu={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
