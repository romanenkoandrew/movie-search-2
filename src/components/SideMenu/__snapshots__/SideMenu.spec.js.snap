// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu component should render SideMenu component 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="SideMenu"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
  collapsed={true}
  collapsible={true}
  css={[Function]}
  onCollapse={[Function]}
  theme="dark"
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="SideMenu"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
    css={[Function]}
    mode="inline"
    theme="dark"
  >
    <MenuItem
      icon={
        <ForwardRef(HomeFilled)
          style={
            Object {
              "color": "green",
            }
          }
        />
      }
      key="/"
    >
      <Link
        to="/"
      >
        Main
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(ProfileFilled)
          style={
            Object {
              "color": "#ff9900",
            }
          }
        />
      }
      key="/watchlist"
    >
      <Link
        to="/watchlist"
      >
        Watch List
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(PlaySquareFilled)
          style={
            Object {
              "color": "red",
            }
          }
        />
      }
      key="/viewedlist"
    >
      <Link
        to="/viewedlist"
      >
        Viewed
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(StarFilled)
          style={
            Object {
              "color": "#ecec20",
            }
          }
        />
      }
      key="/favoritelist"
    >
      <Link
        to="/favoritelist"
      >
        Favorite
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(RestFilled)
          style={
            Object {
              "color": "white",
            }
          }
        />
      }
      key="/blacklist"
    >
      <Link
        to="/blacklist"
      >
        Black List
      </Link>
    </MenuItem>
  </EmotionCssPropInternal>
</EmotionCssPropInternal>
`;

exports[`SideMenu component should render SideMenu component 2`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="SideMenu"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
  collapsed={false}
  collapsible={true}
  css={[Function]}
  onCollapse={[Function]}
  theme="dark"
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="SideMenu"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
    css={[Function]}
    mode="inline"
    theme="dark"
  >
    <MenuItem
      icon={
        <ForwardRef(HomeFilled)
          style={
            Object {
              "color": "green",
            }
          }
        />
      }
      key="/"
    >
      <Link
        to="/"
      >
        Main
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(ProfileFilled)
          style={
            Object {
              "color": "#ff9900",
            }
          }
        />
      }
      key="/watchlist"
    >
      <Link
        to="/watchlist"
      >
        Watch List
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(PlaySquareFilled)
          style={
            Object {
              "color": "red",
            }
          }
        />
      }
      key="/viewedlist"
    >
      <Link
        to="/viewedlist"
      >
        Viewed
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(StarFilled)
          style={
            Object {
              "color": "#ecec20",
            }
          }
        />
      }
      key="/favoritelist"
    >
      <Link
        to="/favoritelist"
      >
        Favorite
      </Link>
    </MenuItem>
    <MenuItem
      icon={
        <ForwardRef(RestFilled)
          style={
            Object {
              "color": "white",
            }
          }
        />
      }
      key="/blacklist"
    >
      <Link
        to="/blacklist"
      >
        Black List
      </Link>
    </MenuItem>
  </EmotionCssPropInternal>
</EmotionCssPropInternal>
`;
